src/message.coffee

```diff
@@ -95,23 +95,39 @@ class SlackTextMessage extends TextMessage
       text = text + "\n" + attachment_text
 
     # Replace links in text async to fetch user and channel info (if present)
-    @replaceLinks(client, text)
-      .then (replacedText) =>
+    mentionFormatting = @replaceLinks(client, text)
+    # Fetch conversation info
+    fetchingConversationInfo = @fetchConversation(client, @_channel_id)
+    Promise.all([mentionFormatting, fetchingConversationInfo])
+      .then (results) =>
+        [ replacedText, conversationInfo ] = results
         text = replacedText
-        text = text.replace /&lt;/g, '<'
-        text = text.replace /&gt;/g, '>'
-        text = text.replace /&amp;/g, '&'
+        text = text.replace /&lt;/g, "<"
+        text = text.replace /&gt;/g, ">"
+        text = text.replace /&amp;/g, "&"
 
+        # Add conversation info to conversation map
+        if not client.channelData[@_channel_id]? and conversationInfo.channel?
```

Comment:
i think adding the conversation info into the conversation map can be done in the same step as fetching the conversation info. in fact, as pointed out below, i think this should be responsibility of the client (the conversation map should be treated as private state and only the client knows that the data is cached).